### PR TITLE
Directory support

### PR DESCRIPTION
**Pull request by Fohlen**
This PR can be merged ASAP. It does not introduce breaking behaviour.

### CODE REVIEW

**Comment by yGuy** [comment]
Please do not advertise this feature too much. It will only lead to troubles because of file system issues (length restrictions, case sensitivity, non-empty directories, etc.) - using this features should be a last resort.

**Comment by Fohlen** [comment]
@yGuy actually we were looking to simplify the workflow in https://github.com/yWorks/yGuard/tree/master/examples/processing with this feature (besides testing from file system, which is much more convenient).

File: docs/task_documentation.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ The `yguard` element has no attributes.
 
 At least one `inoutpair` element or one non-empty `inoutpairs` element has to be specified in order to run the yguard tasks. This element specifies the paths to the input and output jar files.
 
+`inoutpair` also supports the usage of directories. This is detected by [File.isDirectory](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#isDirectory()). However, in general you do not want to use directories for `inoutpairs` (advanced use cases).
+
```
Comment (Fohlen) [inline docs/task_documentation.md:57]: @yGuy does this sound like a proper way to document?

Comment (yGuy) [reply docs/task_documentation.md:57]: LGTM

**Pull request merged**